api/src/opentrons/protocol_api/core/engine/protocol.py

```diff
@@ -190,17 +194,21 @@ def load_labware(
     def move_labware(
         self,
         labware_core: LabwareCore,
-        new_location: Union[DeckSlotName, ModuleCore, NonConnectedModuleCore],
+        new_location: Union[
+            DeckSlotName, ModuleCore, NonConnectedModuleCore, OffDeckType
+        ],
         use_gripper: bool,
         use_pick_up_location_lpc_offset: bool,
         use_drop_location_lpc_offset: bool,
         pick_up_offset: Optional[Tuple[float, float, float]],
         drop_offset: Optional[Tuple[float, float, float]],
     ) -> None:
         """Move the given labware to a new location."""
-        to_location: Union[ModuleLocation, DeckSlotLocation]
+        to_location: Union[ModuleLocation, DeckSlotLocation, str]
         if isinstance(new_location, (ModuleCore, NonConnectedModuleCore)):
             to_location = ModuleLocation(moduleId=new_location.module_id)
+        elif isinstance(new_location, OffDeckType):
+            to_location = OFF_DECK_LOCATION
         else:
             to_location = DeckSlotLocation(slotName=new_location)
```

Comment:
Hmm, although this is not wrong, it works on the assumption that `OffDeckType` value can only be `OFF_DECK`, which is true currently but it would be better to do a thorough check:
```suggestion
        elif new_location == OffDeckType.OFF_DECK:
            to_location = OFF_DECK_LOCATION
        else:
            to_location = DeckSlotLocation(slotName=new_location)
```